AudioPlayerRecorderSamples/src/main/java/com/heavyplayer/audioplayerrecorder/sample/activity/RecorderActivity.java

```diff
@@ -14,65 +27,79 @@
 import android.widget.ListView;
 import android.widget.TextView;
 import android.widget.Toast;
-import com.heavyplayer.audioplayerrecorder.fragment.AudioRecorderFragment;
-import com.heavyplayer.audioplayerrecorder.sample.R;
-import com.heavyplayer.audioplayerrecorder.sample.obj.Item;
-import com.heavyplayer.audioplayerrecorder.utils.AudioUtils;
 
 import java.io.File;
 
-public class RecorderActivity extends ActionBarActivity implements AdapterView.OnItemClickListener {
-	private static Item[] sItems = {
-			new Item(0),
-			new Item(1),
-			new Item(2),
-			new Item(3),
-			new Item(4),
-			new Item(5),
-			new Item(6),
-			new Item(7),
-			new Item(8),
-			new Item(9),
-			new Item(11),
-			new Item(12),
-			new Item(13),
-			new Item(14),
-	};
-
-	protected TextView mTitleView;
-	protected ListView mListView;
+public class RecorderActivity extends AppCompatActivity
+        implements AdapterView.OnItemClickListener, ActivityCompat.OnRequestPermissionsResultCallback {
+    private final static String PERMISSION = Manifest.permission.RECORD_AUDIO;
+    private final static int REQUEST_CODE_PERMISSION = 0;
+
+    private static Item[] sItems = {
+            new Item(0),
+            new Item(1),
+            new Item(2),
+            new Item(3),
+            new Item(4),
+            new Item(5),
+            new Item(6),
+            new Item(7),
+            new Item(8),
+            new Item(9),
+            new Item(11),
+            new Item(12),
+            new Item(13),
+            new Item(14),
+    };
+
+    protected TextView mTitleView;
+    protected ListView mListView;
+
+    // Tracks if permission was just granted,
+    // workaround for https://code.google.com/p/android-developer-preview/issues/detail?id=2823.
+    private boolean mStartRecord;
```

Comment:
I would improve the first doc line to something like "Signals that recording can start, but only after permissions are granted."